x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/ApiKeyBackwardsCompatibilityIT.java

```diff
@@ -424,4 +480,83 @@ private void assertQuery(RestClient restClient, String body, Consumer<List<Map<S
         final List<Map<String, Object>> apiKeys = (List<Map<String, Object>>) responseMap.get("api_keys");
         apiKeysVerifier.accept(apiKeys);
     }
+
+    private boolean nodeSupportsCertificateIdentity(Map<String, Object> nodeDetails) {
+        String nodeVersionString = (String) nodeDetails.get("version");
+        Version nodeVersion = Version.fromString(nodeVersionString);
+        // Certificate identity was introduced in 9.2.0
+        return nodeVersion.onOrAfter(Version.V_9_2_0);
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<Boolean, RestClient> getRestClientByCertificateIdentityCapability() throws IOException {
```

Comment:
The logic this shares with `RolesBackwardsCompatibilityIT` could be moved to `AbstractUpgradeTestCase`

Reply:
I don't see any updates to `RolesBackwardsCompatibilityIT` or `AbstractUpgradeTestCase` was it not possible to put that in shared code? No worries if it doesn't make sense :)